### PR TITLE
Kafka Connect docs review

### PR DESCRIPTION
**Pull request by TomaszGaweda**

### CODE REVIEW

File: docs/modules/integrate/pages/kafka-connect-connectors.adoc [hunk 1]
```diff
@@ -117,4 +118,18 @@ xref:pipelines:submitting-jobs.adoc#submitting-a-job-using-a-java-client-or-embe
 
 == Fault Tolerance
 
-When a Kafka Connect Source connector is deployed to a Hazelcast cluster, snapshots of the connector state are stored in the Jet engine. For example, partition offsets and any metadata required for a restart or recovery. If there is a connector failure, the recovery behavior will depend on the type of connector. Refer to the documentation of your Kafka Connect Source connector for more detailed information. 
+When a Kafka Connect Source connector is deployed to a Hazelcast cluster, snapshots of the connector state are stored by the Jet engine.
```
Comment (rebekah-lawrence) [inline docs/modules/integrate/pages/kafka-connect-connectors.adoc:121]: Passive, recast to something like:

When you deploy a Kafka Connect Source connector to a Hazelcast cluster, the Jet Engine stores snapshots of the connector state.

File: docs/modules/integrate/pages/kafka-connect-connectors.adoc [hunk 2]
```diff
@@ -117,4 +118,18 @@ xref:pipelines:submitting-jobs.adoc#submitting-a-job-using-a-java-client-or-embe
...
+Example for such states are partition offsets and any metadata required for a restart or recovery. If there is a connector failure, the recovery behavior will depend on the type of connector. Refer to the documentation of your Kafka Connect Source connector for more detailed information.
```
Comment (rebekah-lawrence) [inline docs/modules/integrate/pages/kafka-connect-connectors.adoc:122]: Examples of snapshotted connector states include partition offsets and any metadata required for a restart or for recovery. If there is a connector failure, recovery behavior depends on.....

File: docs/modules/integrate/pages/kafka-connect-connectors.adoc [hunk 3]
```diff
@@ -117,4 +118,18 @@ xref:pipelines:submitting-jobs.adoc#submitting-a-job-using-a-java-client-or-embe
...
+
+== Parallelism and reconfigurations
```
Comment (rebekah-lawrence) [inline docs/modules/integrate/pages/kafka-connect-connectors.adoc:124]: initial cap on Reconfigurations

File: docs/modules/integrate/pages/kafka-connect-connectors.adoc [hunk 4]
```diff
@@ -117,4 +118,18 @@ xref:pipelines:submitting-jobs.adoc#submitting-a-job-using-a-java-client-or-embe
...
+
+Kafka Connect Source uses `tasks.max` configuration property to determine parallelism.
```
Comment (rebekah-lawrence) [inline docs/modules/integrate/pages/kafka-connect-connectors.adoc:126]: add the after uses: Kafka Connect Source uses the `task.max`....

File: docs/modules/integrate/pages/kafka-connect-connectors.adoc [hunk 5]
```diff
@@ -117,4 +118,18 @@ xref:pipelines:submitting-jobs.adoc#submitting-a-job-using-a-java-client-or-embe
...
+Even if pipeline's total parallelism is higher than `tasks.max` value, the `tasks.max` value takes precedence.
```
Comment (rebekah-lawrence) [inline docs/modules/integrate/pages/kafka-connect-connectors.adoc:127]: Reverse: The `tasks.max` value takes precedence, even when a pipeline's total parallelism is higher than that value.

File: docs/modules/integrate/pages/kafka-connect-connectors.adoc [hunk 6]
```diff
@@ -117,4 +118,18 @@ xref:pipelines:submitting-jobs.adoc#submitting-a-job-using-a-java-client-or-embe
...
+It is prohibited to have pipeline's total parallelism (edge's local parallelism * number of nodes) lower than `tasks.max`.
```
Comment (rebekah-lawrence) [inline docs/modules/integrate/pages/kafka-connect-connectors.adoc:128]: A pipeline's total parallelism cannot be lower than the `tasks.max` value.

I'd suggest putting the calculation under the sentence on line 127, along the lines of:

A pipleine's total parallelism is calculated as follows:

    local parallelism of edge * number of nodes

File: docs/modules/integrate/pages/kafka-connect-connectors.adoc [hunk 7]
```diff
@@ -117,4 +118,18 @@ xref:pipelines:submitting-jobs.adoc#submitting-a-job-using-a-java-client-or-embe
...
+
+After creation of an instance of a SourceConnector, Hazelcast Jet asks it to provide `tasks.max` task configurations. If the
```
Comment (rebekah-lawrence) [inline docs/modules/integrate/pages/kafka-connect-connectors.adoc:130]: Hazelcast Jet requests the `task.max` task configuration when an instance of a Source Connector has been created, .

File: docs/modules/integrate/pages/kafka-connect-connectors.adoc [hunk 8]
```diff
@@ -117,4 +118,18 @@ xref:pipelines:submitting-jobs.adoc#submitting-a-job-using-a-java-client-or-embe
...
+SourceConnector returns less than `tasks.max` configs, the remaining processors will become no-op.
```
Comment (rebekah-lawrence) [inline docs/modules/integrate/pages/kafka-connect-connectors.adoc:131]: If the Source Connector returns ????, the remaining processors are ignored.

???? I wasn't sure what "less than `task.max` configs" meant; is it:

a value lower than the current `tasks.max` configuration

File: docs/modules/integrate/pages/kafka-connect-connectors.adoc [hunk 9]
```diff
@@ -117,4 +118,18 @@ xref:pipelines:submitting-jobs.adoc#submitting-a-job-using-a-java-client-or-embe
...
+
+When SourceConnector requests a reconfiguration, Hazelcast asks for a new set of configs and distributes them across the cluster.
```
Comment (rebekah-lawrence) [inline docs/modules/integrate/pages/kafka-connect-connectors.adoc:133]: When a Source connector requests reconfiguration, Hazelcast requests the new configuration and distributes it across the cluster.

File: docs/modules/integrate/pages/kafka-connect-connectors.adoc [hunk 10]
```diff
@@ -117,4 +118,18 @@ xref:pipelines:submitting-jobs.adoc#submitting-a-job-using-a-java-client-or-embe
...
+Existing processors will be restarted automatically with new configuration assigned. If new number of returned task configurations
```
Comment (rebekah-lawrence) [inline docs/modules/integrate/pages/kafka-connect-connectors.adoc:134]: Existing processors automatically restart to use the updated configuration. If supported by the new configuration, previously-ignored processor entries will be processed.

**Review by rebekah-lawrence — CHANGES_REQUESTED** [review]

File: docs/modules/integrate/pages/kafka-connect-connectors.adoc (hunk 8 above)
Comment (TomaszGaweda) [reply docs/modules/integrate/pages/kafka-connect-connectors.adoc:131]: >the remaining processors are ignored.
That wouldn't be true IMHO; processors are not ignored, they are running and waiting for reconfiguration, but main processing method will become no-op (will never return anything unless a reconfiguration is requested).

> I wasn't sure what "less than task.max configs" meant; is it:

Will reword this to:
```
If the SourceConnector returns fewer configurations than `tasks.max` value, the remaining processors will become no-op.
```

File: docs/modules/integrate/pages/kafka-connect-connectors.adoc (hunk 9 above)
Comment (TomaszGaweda) [reply docs/modules/integrate/pages/kafka-connect-connectors.adoc:133]: "a new set of configs" is intentional; the source connector returns as many distinct configurations as user requests

File: docs/modules/integrate/pages/kafka-connect-connectors.adoc (hunk 10 above)
Comment (TomaszGaweda) [reply docs/modules/integrate/pages/kafka-connect-connectors.adoc:134]: >previously-ignored processor entries will be processed.

There are no processor entries ;) Just processors which were no-op (retuning nothing all the time), that start to produce data

File: docs/modules/integrate/pages/kafka-connect-connectors.adoc (hunk 9 above)
Comment (rebekah-lawrence) [reply docs/modules/integrate/pages/kafka-connect-connectors.adoc]: And the configuration sets make up the new configuration - in most circumstances, the plural of configuration is configuration. 

Can you clarify what you mean by distinct configurations? Multiple config files/changed lines within a single config file/both of these? That might help pin down the right phrase....

File: docs/modules/integrate/pages/kafka-connect-connectors.adoc [hunk 11]
```diff
@@ -117,4 +118,21 @@ xref:pipelines:submitting-jobs.adoc#submitting-a-job-using-a-java-client-or-embe
 
 == Fault Tolerance
 
-When a Kafka Connect Source connector is deployed to a Hazelcast cluster, snapshots of the connector state are stored in the Jet engine. For example, partition offsets and any metadata required for a restart or recovery. If there is a connector failure, the recovery behavior will depend on the type of connector. Refer to the documentation of your Kafka Connect Source connector for more detailed information. 
+When you deploy a Kafka Connect Source connector to a Hazelcast cluster, the Jet engine stores snapshots of the connector state.
+Examples of snapshotted connector states include partition offsets and any metadata required for a restart or for recovery. If there is a connector failure, the recovery behavior will depend on the type of connector. Refer to the documentation of your Kafka Connect Source connector for more detailed information.
```
Comment (rebekah-lawrence) [inline docs/modules/integrate/pages/kafka-connect-connectors.adoc:122]:
```suggestion
Examples of snapshotted connector states include partition offsets and any metadata required for a restart or for recovery. If there is a connector failure, the recovery behavior depends on the type of connector. Refer to the documentation of your Kafka Connect Source connector for more detailed information.
```

File: docs/modules/integrate/pages/kafka-connect-connectors.adoc [hunk 12]
```diff
@@ -117,4 +118,21 @@ xref:pipelines:submitting-jobs.adoc#submitting-a-job-using-a-java-client-or-embe
...
+
+Kafka Connect Source uses the `tasks.max` configuration property to determine parallelism.
+The `tasks.max` value takes precedence even if pipeline's total parallelism is higher than that value.
+A pipeline's total parallelism cannot be lower than `tasks.max`, where total parallelism is calculated as:
+```
+total parallelism = edge's local parallelism * number of nodes
+```
+
+Hazelcast Jet requests `tasks.max` task configurations after an instance of SourceConnector has been created. If the
+SourceConnector returns fewer configurations than `tasks.max` value, the remaining processors will become no-op.
```
Comment (rebekah-lawrence) [inline docs/modules/integrate/pages/kafka-connect-connectors.adoc:134]:
```suggestion
SourceConnector returns fewer configurations than specified by the `tasks.max` value, the remaining processors do nothing.
```

File: docs/modules/integrate/pages/kafka-connect-connectors.adoc [hunk 13]
```diff
@@ -117,4 +118,21 @@ xref:pipelines:submitting-jobs.adoc#submitting-a-job-using-a-java-client-or-embe
...
+
+When SourceConnector requests a reconfiguration, Hazelcast requests for a new set of configurations and distributes them across the cluster.
+Existing processors automatically restart to use the updated configuration. If new number of returned task configurations
```
Comment (rebekah-lawrence) [inline docs/modules/integrate/pages/kafka-connect-connectors.adoc:137]:
```suggestion
Existing processors automatically restart and use the updated configuration. If the number of task configuration sets 
```

**Review by rebekah-lawrence — CHANGES_REQUESTED** [review]

File: docs/modules/integrate/pages/kafka-connect-connectors.adoc [hunk 14]
```diff
@@ -117,4 +118,21 @@ xref:pipelines:submitting-jobs.adoc#submitting-a-job-using-a-java-client-or-embe
...
+The `tasks.max` value takes precedence even if pipeline's total parallelism is higher than that value.
+A pipeline's total parallelism cannot be lower than `tasks.max`, where total parallelism is calculated as:
+```
+total parallelism = edge's local parallelism * number of nodes
+```
+
+Hazelcast Jet requests `tasks.max` task configurations after an instance of SourceConnector has been created. If the
+SourceConnector returns fewer configurations than `tasks.max` value, the remaining processors will become no-op.
+
+When SourceConnector requests a reconfiguration, Hazelcast requests for a new set of configurations and distributes them across the cluster.
```
[the remark this answers is not in the document]
Comment (TomaszGaweda) [reply docs/modules/integrate/pages/kafka-connect-connectors.adoc]: I changed it to "the new collection of configuration sets"

I forgot that "configuration set" has already a meaning, while I meant set like in math :D Hopefully now it's ok

**Review by rebekah-lawrence — APPROVED** [review]
LGTM

Thanks for bearing with me!

File: docs/modules/integrate/pages/kafka-connect-connectors.adoc (hunk 9 above)
Comment (TomaszGaweda) [reply docs/modules/integrate/pages/kafka-connect-connectors.adoc]: Just for the record (I commented elsewhere): I meant a collection of configurations (set in terms of math), I just went full 3yo with my English and forgot about "configuration set" in other meaning.

**Comment by TomaszGaweda** [comment]
Thanks @rebekah-lawrence for detailed review!

**Review by orcunc — APPROVED** [review]
